Pod/Classes/PINRemoteImageManager.m

```diff
@@ -1121,6 +1177,15 @@ - (void)didReceiveData:(NSData *)data forTask:(NSURLSessionDataTask *)dataTask
                 task.progressImage.progressThresholds = self.progressThresholds;
             }
         }
+    
+        for (PINRemoteImageCallbacks *completion in task.callbackBlocks.allValues) {
```

Comment:
Do you think it might be better to get a copy of the callbacks, and then do the actual callbacks outside of the lock? That way, blocking the callback won't lock up PINRemoteImage.